components/automate-ui/src/app/pages/policy/add-members/policy-add-members.component.ts

```diff
@@ -334,15 +334,14 @@ export class PolicyAddMembersComponent implements OnInit, OnDestroy {
         if (formValues.type === 'user' || formValues.type === 'team') {
           this.addIdentityControl();
         } else if (formValues.type === 'token') {
-          this.addNameControl();
+          this.addNameControl(formValues.identityProvider);
         }
         break;
       case 'identityProvider':
         this.resetErrors();
+        this.expressionForm.removeControl('name');
```

Comment:
We now need to remove the name control, so that we can add a new name control incase it has changed.  I think we're better off just removing and then adding instead of checking to see if we should update it if it already exists.  This will remove some unnecessary logic.